python/ray/tune/result_grid.py

```diff
@@ -232,7 +254,8 @@ def _trial_to_result(self, trial: Trial) -> Result:
             checkpoint=checkpoint,
             metrics=trial.last_result.copy(),
             error=self._populate_exception(trial),
-            log_dir=Path(trial.local_path) if trial.local_path else None,
+            _local_path=str(Path(trial.local_path)) if trial.local_path else None,
```

Comment:
`str(Path(...))` is to normalize the path string?

Reply:
`log_dir` was a `pathlib.Path` but because all our other `path` properties are strings, this should be a string, too

Reply:
Got it. Is there a reason we convert to a Path then back to a string here?